### PR TITLE
glibc: update to 2.38. 

### PR DESCRIPTION
**Pull request by oreo639**
#### Testing the changes
- I tested the changes in this PR: **briefly**

The following packages must be removed from the repo before merging this PR:
- [x] php8.0-apache-8.0.29_1: libcrypt.so.1 (https://repo-default.voidlinux.org/current)
- [x] php8.0-embed-8.0.29_1: libcrypt.so.1 (https://repo-default.voidlinux.org/current)
- [x] burp-client-1.4.40_12: libcrypt.so.1 (https://repo-default.voidlinux.org/current)
- [x] user-manager-5.19.5_1: libcrypt.so.1 (https://repo-default.voidlinux.org/current)

[ci skip]
https://sourceware.org/glibc/wiki/Release/2.38
glibc 2.38 disables libcrypt by default with a recommendation for distros to switch to [libxcrypt](https://github.com/besser82/libxcrypt) (as they intend to remove it entirely).
libxcrypt can be built as either libcrypt.so.1 or libcrypt.so.2.
Debian and SUSE only build libcrypt.so.1.
Arch Linux and Fedora build libcrypt.so.2 along with a libxcrypt-compat package.
The rationale for that decision is explained here:
https://fedoraproject.org/wiki/Changes/FullyRemoveDeprecatedAndUnsafeFunctionsFromLibcrypt
https://fedoraproject.org/wiki/Changes/libcrypt_so_1_Let_encrypt_encrypt_r_setkey_setkey_r_and_fcrypt_return_ENOSYS_instead_of_performing_any_real_operation

My inclination is to build libcrypt.so.2 with a libxcrypt-compat package without enosys but I am open to handling it differently.
Keep in mind that packages compiled against libxcrypt's libcrypt.so.1 will not run with glibc's libcrypt.so.1 (but applications compiled with glibc's libcrypt.so.1 will run with libxcrypt's libcrypt.so.1).

You can get the packages using libcrypt using `xbps-query -Rs libcrypt.so -p shlib-requires`.

### CODE REVIEW

**Comment by oreo639** [comment]
> Either the template provides the -compat package or it doesn't.

Yeah. I had it because I was experimenting with musl where it would be completely useless. I removed the build option.

**Comment by oreo639** [comment]
I found one thing that requires libcrypt.so.1 (XAMPP).

**Comment by classabbyamp** [comment]
should we be rebuilding the cross toolchains before everything else?

**Comment by oreo639** [comment]
perl will need to be rebuilt before the cross toolchains, but aside from that it should be fine to push it later afaik.

**Comment by classabbyamp** [comment]
can you rebase and reorder the commits so that it's all in order to merge in chunks? I'd like to push up to the cross toolchains before the mass rebuild for crypt

**Comment by oreo639** [comment]
Alright.

**Comment by leahneukirchen** [comment]
Sorry, I only got around reviewing this now... I think this needs more changes.

1) I'd really like to avoid perl-bootstrap.  The libxcrypt release tarballs contain most of the Perl generated headers except for the list of enabled hashes... I think those we could keep in files/ (this will require a small patch against libxcrypt configure.)

2) IMO libxcrypt-devel should be in base-devel for glibc, changing all templates that need libcrypt is not required then (but they should be revbumped for rebuild).

**Comment by leahneukirchen** [comment]
Yeah you're right re perl.

Also I meant to say xcrypt-devel should go into base-chroot.

**Comment by oreo639** [comment]
Alright.

File: common/shlibs [hunk 1]
```diff
@@ -43,13 +43,39 @@ libnss_dns.so.2 glibc-2.36_1
...
-libmemusage.so glibc-2.36_1
-libSegFault.so glibc-2.36_1
-libpcprofile.so glibc-2.36_1
-libcidn.so.1 glibc-2.36_1
-libmvec.so.1 glibc-2.36_1
+libmemusage.so glibc-2.38_1
+libSegFault.so glibc-2.38_1
+libpcprofile.so glibc-2.38_1
+libcidn.so.1 glibc-2.38_1
+libmvec.so.1 glibc-2.3_1
```
Comment (leahneukirchen) [inline common/shlibs:78]: Typo: 8 missing

File: srcpkgs/glibc/template [hunk 2]
```diff
@@ -98,14 +96,10 @@ do_configure() {
 	SHELL=/bin/bash ../configure ${configure_args} \
 		--bindir=/usr/bin --sbindir=/usr/bin \
 		--libdir=${_libdir} --libexecdir=${_libdir} \
-		--enable-add-ons=libidn \
 		--enable-multi-arch --enable-bind-now \
-		--disable-profile --enable-kernel=3.2.0 \
-		--enable-stack-guard-randomization \
-		--without-selinux --without-cvs --without-gd \
-		--disable-lock-elision \
```
Comment (leahneukirchen) [inline srcpkgs/glibc/template:106]: Is lock elision disabled by default now or why was this removed?

Comment (oreo639) [reply srcpkgs/glibc/template:106]: https://github.com/bminor/glibc/commit/07ed18d26a342741cb25a4739158c65ed9dd4d09

elison tunables was added which depends on lock elison so the ability to disable lock elison was removed in 2017.

File: common/shlibs (hunk 1 above)
Comment (oreo639) [reply common/shlibs:78]: Thank you.
